tests/kyber-armv7m/main.c

```diff
@@ -0,0 +1,1376 @@
+MAKE_BENCH_frombytes_mul_32(kyber_frombytes_mul_16_32_opt_m7, frombytes_mul_asm_16_32_opt_m7_wrap);
+
+MAKE_BENCH_frombytes_mul_32(kyber_frombytes_mul_acc_32_32, frombytes_mul_asm_acc_32_32_wrap);
+MAKE_BENCH_frombytes_mul_32(kyber_frombytes_mul_acc_32_32_opt_m7, frombytes_mul_asm_acc_32_32_opt_m7_wrap);
+
+MAKE_BENCH_frombytes_mul_32_16(kyber_frombytes_mul_acc_32_16, frombytes_mul_asm_acc_32_16_wrap);
+MAKE_BENCH_frombytes_mul_32_16(kyber_frombytes_mul_acc_32_16_opt_m7, frombytes_mul_asm_acc_32_16_opt_m7_wrap);
+
+MAKE_BENCH_3(kyber_basemul_asm_pqm4,basemul_asm_wrap)
+MAKE_BENCH_3(kyber_basemul_asm_pqm4_opt_m7,basemul_asm_opt_m7_wrap)
+
+MAKE_BENCH_3(kyber_basemul_asm_acc_pqm4,basemul_asm_acc_wrap)
+MAKE_BENCH_3(kyber_basemul_asm_acc_pqm4_opt_m7,basemul_asm_acc_opt_m7_wrap)
+
+MAKE_BENCH_frombytes_mul(kyber_frombytes_mul_pqm4,frombytes_mul_asm_wrap)
+MAKE_BENCH_frombytes_mul(kyber_frombytes_mul_pqm4_opt_m7,frombytes_mul_asm_opt_m7_wrap)
+
+MAKE_BENCH_frombytes_mul(kyber_frombytes_mul_acc_pqm4,frombytes_mul_asm_acc_wrap)
+MAKE_BENCH_frombytes_mul(kyber_frombytes_mul_acc_pqm4_opt_m7,frombytes_mul_asm_acc_opt_m7_wrap)
+
+MAKE_BENCH_matacc(kyber_matacc_asm_pqm4,matacc_asm)
+MAKE_BENCH_matacc(kyber_matacc_asm_pqm4_opt_m7,matacc_asm_opt_m7)
+
+MAKE_BENCH_matacc(kyber_matacc_asm_acc_pqm4,matacc_asm_acc)
+MAKE_BENCH_matacc(kyber_matacc_asm_acc_pqm4_opt_m7,matacc_asm_acc_opt_m7)
+
+MAKE_BENCH_matacc_cache_32(kyber_matacc_asm_cache_16_32_pqm4,matacc_asm_cache_16_32)
+MAKE_BENCH_matacc_cache_32(kyber_matacc_asm_cache_16_32_pqm4_opt_m7,matacc_asm_cache_16_32_opt_m7)
+
+MAKE_BENCH_matacc_cache_32(kyber_matacc_asm_cache_32_32_pqm4,matacc_asm_cache_32_32)
+MAKE_BENCH_matacc_cache_32(kyber_matacc_asm_cache_32_32_pqm4_opt_m7,matacc_asm_cache_32_32_opt_m7)
+
+MAKE_BENCH_matacc_cache_32_16(kyber_matacc_asm_cache_32_16_pqm4,matacc_asm_cache_32_16)
+MAKE_BENCH_matacc_cache_32_16(kyber_matacc_asm_cache_32_16_pqm4_opt_m7,matacc_asm_cache_32_16_opt_m7)
+
+MAKE_BENCH_matacc_opt_32(kyber_matacc_asm_opt_16_32_pqm4,matacc_asm_opt_16_32)
+MAKE_BENCH_matacc_opt_32(kyber_matacc_asm_opt_16_32_pqm4_opt_m7,matacc_asm_opt_16_32_opt_m7)
+
+MAKE_BENCH_matacc_opt_32(kyber_matacc_asm_opt_32_32_pqm4,matacc_asm_opt_32_32)
+MAKE_BENCH_matacc_opt_32(kyber_matacc_asm_opt_32_32_pqm4_opt_m7,matacc_asm_opt_32_32_opt_m7)
+
+MAKE_BENCH_matacc_opt_32_16(kyber_matacc_asm_opt_32_16_pqm4,matacc_asm_opt_32_16)
+MAKE_BENCH_matacc_opt_32_16(kyber_matacc_asm_opt_32_16_pqm4_opt_m7,matacc_asm_opt_32_16_opt_m7)
+
+
+int main(void)
+{
+    int ret = 0;
+    debug_test_start( "\nKyber All Test!\n" );
+
+    /* Test cases */
+    if( test_ntt_pqm4() != 0 ){return( 1 );}
+    if( test_ntt_pqm4_opt() != 0 ){return( 1 );}
+    if( test_ntt_acle() != 0){return( 1 );}
+
+    if( test_invntt_pqm4() != 0 ){return( 1 );}
+    if( test_invntt_pqm4_opt() != 0 ){return( 1 );}
+    // TODO: test currently fails -- need to debug
+    // if( test_invntt_acle() != 0 ){return( 1 );}
```

Comment:
Please remove this one - I don't think we need this. 
Maybe remove everything using ACLE? It didn't turn out to be competitive.

Reply:
I removed it.